### PR TITLE
Implement -fsyntax-only

### LINKED ISSUE #8: Debug mode builds of Flang should properly handle -fsyntax-only option
The -fsyntax-only option is very popular for testing clang. When applied to debug mode build of Flang it causes assertion fail:
```
tools/clang/lib/Frontend/FrontendAction.cpp:265: bool clang::FrontendAction::BeginSourceFile(clang::CompilerInstance&, const clang::FrontendInputFile&): Assertion `hasIRSupport() && "This action does not have IR file support!"' failed.
```
Does not occur with Release mode builds of Flang.

### LINKED ISSUE #33: AddFlangSystemIncludeArgs() second argument name inconsistency
CC1Args vs F901Args

Linux.h:
```
  AddFlangSystemIncludeArgs(const llvm::opt::ArgList &DriverArgs,
                            llvm::opt::ArgStringList &CC1Args) const override;
```
Linux.cpp:
```
void Linux::AddFlangSystemIncludeArgs(const ArgList &DriverArgs,
                                      ArgStringList &F901Args) const
```

### PR DESCRIPTION
**Pull request by ppenzin-nv**
Also fix argument mismatch in lib/Driver/ToolChains.h

Fixes #33 
Fixes #8

### CODE REVIEW

File: lib/Driver/Tools.cpp [hunk 1]
```diff
@@ -4124,6 +4124,8 @@ void FlangFrontend::ConstructJob(Compilation &C, const JobAction &JA,
   ArgStringList CommonCmdArgs;
   ArgStringList UpperCmdArgs;
   ArgStringList LowerCmdArgs;
+  SmallString<256> Stem;
```
Comment (sscalpone) [inline lib/Driver/Tools.cpp:4127]: Why not std:string??

**Comment by sscalpone** [comment]
Do you want to reference issue #33?

**Comment by sscalpone** [comment]
Do you want to reference issue #33?

File: lib/Driver/Tools.cpp (hunk 1 above)
Comment (ppenzin-nv) [reply lib/Driver/Tools.cpp:4127]: for llvm::sys::path::replace_extension

Comment (sscalpone) [reply lib/Driver/Tools.cpp:4127]: Got it. Thanks

**Pull request merged**